windows/setup.ps1

```diff
@@ -77,29 +81,79 @@ foreach ($adapter in $ActiveNetcfg)
 Set-Attr $result.ansible_facts "ansible_distribution" $osversion.VersionString
 Set-Attr $result.ansible_facts "ansible_distribution_version" $osversion.Version.ToString()
+Set-Attr $result.ansible_facts "ansible_distribution_major_version" $osversion.Version.Major.ToString()
+Set-Attr $result.ansible_facts "ansible_kernel" $osversion.Version.ToString()
+
+Set-Attr $result.ansible_facts "ansible_machine_id" $user.User.AccountDomainSid.Value
+Set-Attr $result.ansible_facts "ansible_domain" $ip_props.DomainName
+Set-Attr $result.ansible_facts "ansible_nodename" ($ip_props.HostName + "." + $ip_props.DomainName)
+Set-Attr $result.ansible_facts "ansible_windows_domain" $win32_cs.Domain
 
+Set-Attr $result.ansible_facts "ansible_owner_name" ([string] $win32_cs.PrimaryOwnerName)
+Set-Attr $result.ansible_facts "ansible_owner_contact" ([string] $win32_cs.PrimaryOwnerContact)
+
+Set-Attr $result.ansible_facts "ansible_user_dir" $env:userprofile
+Set-Attr $result.ansible_facts "ansible_user_gecos" ([string] $user.Label)
+Set-Attr $result.ansible_facts "ansible_user_id" $env:username
+Set-Attr $result.ansible_facts "ansible_user_uid" ([int] $user.User.Value.Substring(42))
+Set-Attr $result.ansible_facts "ansible_user_sid" $user.User.Value
+
+# Use English locale
+[System.Threading.Thread]::CurrentThread.CurrentCulture = [System.Globalization.CultureInfo]::GetCultureInfo('en-US')
 $date = New-Object psobject
-Set-Attr $date "date" (Get-Date -format d)
-Set-Attr $date "year" (Get-Date -format yyyy)
-Set-Attr $date "month" (Get-Date -format MM)
-Set-Attr $date "day" (Get-Date -format dd)
-Set-Attr $date "hour" (Get-Date -format HH)
-Set-Attr $date "minute" (Get-Date -format mm)
-Set-Attr $date "iso8601" (Get-Date -format s)
+$datetime = (Get-Date)
+$datetime_utc = $datetime.ToUniversalTime()
+Set-Attr $date "date" $datetime.ToString("yyyy-MM-dd")
+Set-Attr $date "day" $datetime.ToString("dd")
+Set-Attr $date "epoch" (Get-Date -UFormat "%s")
+Set-Attr $date "hour" $datetime.ToString("HH")
+Set-Attr $date "iso8601" $datetime_utc.ToString("yyyy-MM-ddTHH:mm:ssZ")
+Set-Attr $date "iso8601_basic" $datetime.ToString("yyyyMMddTHHmmssffffff")
+Set-Attr $date "iso8601_basic_short" $datetime.ToString("yyyyMMddTHHmmss")
+Set-Attr $date "iso8601_micro" $datetime_utc.ToString("yyyy-MM-ddTHH:mm:ss.ffffffZ")
+Set-Attr $date "minute" $datetime.ToString("mm")
+Set-Attr $date "month" $datetime.ToString("MM")
+Set-Attr $date "second" $datetime.ToString("ss")
+Set-Attr $date "time" $datetime.ToString("HH:mm:ss")
+Set-Attr $date "tz_offset" $datetime.ToString("zzzz")
+Set-Attr $date "tz" ([System.TimeZoneInfo]::Local.Id)
+Set-Attr $date "weekday" $datetime.ToString("dddd")
+Set-Attr $date "weekday_number" (Get-Date -UFormat "%w")
+Set-Attr $date "weeknumber" (Get-Date -UFormat "%W")
+Set-Attr $date "year" $datetime.ToString("yyyy")
 Set-Attr $result.ansible_facts "ansible_date_time" $date
 
-Set-Attr $result.ansible_facts "ansible_totalmem" $capacity
```

Comment:
Lemme check with a couple others- since we haven't written it down yet (I was hoping to get to windows setup return docs for 2.2), it's probably ok...
@trondhindenes @jhawkesworth @bcoca any opinion?